intranet/apps/dashboard/views.py

```diff
@@ -255,39 +256,43 @@ def get_announcements_list(request, context):
             midnight = timezone.localtime().replace(hour=0, minute=0, second=0, microsecond=0)
             events = Event.objects.visible_to_user(user).filter(time__gte=midnight, show_on_dashboard=True)
 
-    items = sorted(chain(announcements, events), key=lambda item: (item.pinned, item.added))
-    items.reverse()
+    def announcements_sorting_key(item: Announcement):
+        nonlocal context
```

Comment:
nonlocal isn't necessary here. You only need nonlocal / global if you're assigning a value to the variable (since python thinks your defining a new variable).

Reply:
Good to know!